plugins/scaffolder-backend-module-bitbucket-server/src/actions/bitbucketServerPullRequest.ts

```diff
@@ -401,7 +405,8 @@ export function createPublishBitbucketServerPullRequestAction(options: {
           startPoint: latestCommit,
         });
 
-        const remoteUrl = `https://${host}/scm/${project}/${repo}.git`;
+        const protocol = parseProtocol(apiBaseUrl);
+        const remoteUrl = `${protocol}://${host}/scm/${project}/${repo}.git`;
```

Comment:
If the use case only intends to use `http` or `https`, it would be better to use a boolean as a parameter.
Allowing to prepend any string in front will enable the execution of malicious URL injections.

Nothing prevents someone instead of `http` or `https` to provide here: `https://iwillhuckyou.now///${host}/scm/${project}/${repo}.git`

Reply:
That shouldn't be possible, the URL object is parsing out the protocol scheme :

- https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol

Here's a working example:

- https://playcode.io/2017786

Reply:
@acierto I updated to use a boolean for whether to use "http" or "https" as a protocol scheme.  leaving the default to be "https"